pkg/arvo/app/invite-hook.hoon

```diff
@@ -1,123 +1,105 @@
-        %json
-      ::  only accept json from ourselves.
-      ::
-      ?>  (team:title our.bowl src.bowl)
-      =/  act  (json-to-action !<(json vase))
-      ?>  ?=(%invite -.act)
-      [(invite-hook-poke:do recipient.invite.act act)]~
-    ::
-        %invite-action
-      =/  act=invite-action  !<(invite-action vase)
-      ?.  ?=(%invite -.act)  ~
-      ?:  (team:title our.bowl src.bowl)
-        ::  outgoing. we must be inviting another ship. send them the invite.
-        ::
-        ?<  (team:title our.bowl recipient.invite.act)
-        [(invite-hook-poke:do recipient.invite.act act)]~
-      ::  else incoming. ensure invitatory exists and invite is not a duplicate.
-      ::
-      ?>  ?=(^ (invitatory-scry:do path.act))
-      ?>  ?=(~ (invite-scry:do path.act uid.act))
-      [(invite-poke:do path.act act)]~
-    ==
-  ::
-  ++  on-peek   on-peek:def
-  ++  on-watch  on-watch:def
-  ++  on-leave  on-leave:def
-  ++  on-agent  on-agent:def
-  ++  on-arvo   on-arvo:def
-  ++  on-fail   on-fail:def
-  --
 ::
 |_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
 ::
-++  invite-hook-poke
-  |=  [=ship action=invite-action]
-  ^-  card
-  :*  %pass
-      /invite-hook
-      %agent
-      [ship %invite-hook]
-      %poke
-      %invite-action
-      !>(action)
-  ==
+++  on-init  [~ this]
+++  on-save  !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  [~ this(state !<(state-0 old))]
 ::
-++  invite-poke
-  |=  [=path action=invite-action]
-  ^-  card
-  :*  %pass
-      path
-      %agent
-      [our.bowl %invite-store]
-      %poke
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  :_  this
+  ?+  mark  (on-poke:def mark vase)
       %invite-action
-      !>(action)
+    =/  act=invite-action  !<(invite-action vase)
+    ?.  ?=(%invite -.act)  ~
+    ?:  (team:title our.bowl src.bowl)
+      ::  outgoing. we must be inviting another ship. send them the invite.
+      ::
+      ?<  (team:title our.bowl recipient.invite.act)
+      [(invite-hook-poke recipient.invite.act act)]~
+    ::  else incoming. ensure invitatory exists and invite is not a duplicate.
+    ::
+    ?>  ?=(^ (invitatory-scry path.act))
+    ?>  ?=(~ (invite-scry path.act uid.act))
+    [(invite-poke path.act act)]~
   ==
-::
-++  invitatory-scry
-  |=  pax=path
-  ^-  (unit invitatory)
-  =.  pax
-    ;:  weld
-      /(scot %p our.bowl)/invite-store/(scot %da now.bowl)/invitatory
-      pax
-      /noun
+  ::
+  ++  invite-hook-poke
+    |=  [=ship action=invite-action]
+    ^-  card
+    :*  %pass
+        /invite-hook
+        %agent
+        [ship %invite-hook]
+        %poke
+        %invite-action
+        !>(action)
     ==
-  .^((unit invitatory) %gx pax)
-::
-++  invite-scry
-  |=  [pax=path uid=serial]
-  ^-  (unit invite)
-  =.  pax
-    ;:  weld
-      /(scot %p our.bowl)/invite-store/(scot %da now.bowl)/invite
-      pax
-      /(scot %uv uid)/noun
+  ::
+  ++  invite-poke
```

Comment:
Kinda think that these should maybe be moved to a `lib/invite.hoon` file, because we poke the invite-hook in a bunch of places elsewhere in userspace.

Reply:
Good idea